docs/modules/candid-guide/index.adoc

```diff
@@ -23,84 +27,99 @@ service counter : {
- * Floating point types (`float32`, `float64`)
- * The Boolean type (`bool`)
- * Types for textual (`text`) and binary (`blob`) data
- * Container types, including variants (`opt`, `vec`, `record`, `variant`)
- * Reference types (`service`, `func`, `principal`)
- * The special `null`, `reserved` and `empty` types
+ * Unbounded integral number types (`+nat+`, `+int+`).
+ * Bounded integral number (`+nat8+`,`+nat16+`, `+nat32+`, `+nat64+`, `+int8+`,`+int16+`, `+int32+`, `+int64+`).
+ * Floating point types (`+float32+`, `+float64+`).
+ * The Boolean type (`+bool+`).
+ * Types for textual (`+text+`) and binary (`+blob+`) data.
+ * Container types, including variants (`+opt+`, `+vec+`, `+record+`, `+variant+`).
+ * Reference types (`+service+`, `+func+`, `+principal+`).
+ * The special `+null+`, `+reserved+` and `+empty+` types.
 
-All types are described in detail in the <<reference, reference section>>.
+All types are described in detail in the xref:candid-ref{outfilesuffix}[Reference] section.
 
-The philosophy behind this set of types is that they are sufficient to describe the _structure_ of data, so that information can be encoded, passed around and decoded, but intentionally do not describe _semantic_ constraints beyond what’s needed to describe the representation (e.g. no way to express that a number should be even, that a vector has a certain length, or that the elements of a vector are sorted).
+The philosophy behind this set of types is that they are sufficient to describe the _structure_ of data, so that information can be encoded, passed around and decoded, but intentionally do not describe _semantic_ constraints beyond what’s needed to describe the representation. For example, there's no way to express that a number should be even, that a vector has a certain length, or that the elements of a vector are sorted.
 
-Furthermore, the set of types is chosen to allow a natural mapping to the respective host language (Motoko, Rust, JavaScript, etc.), settling on reasonable, canonical choices.
+Candid supports this set of types to allow a natural mapping of data types based on reasonable, canonical choices suitable for each host language, whether you are writing your code in Motoko, Rust, JavaScript, or some other language.
 
 [#candid-service-descriptions]
-=== Candid service descriptions
+== Candid service descriptions
 
-With these Candid types, we now can describe a service. Such a service description file can be written by hand, but typically it is generated from a service implementation, as described later.
+Once you are familiar with the Candid types, you can use them describe a service. A Candid service description file—a `+.DID+` file—can either be written by hand or generated from a service implementation.
+
+Before you explore how to generate a service description file for a specific host language, let's take a closer look at the structure of the file and its constituent parts.
```

Comment:
I'd avoid most references to "files", since they are just a way in which such a description can be physically stored, but on the IC they may occur in other form as well.
```suggestion
Before you explore how to generate a service descriptions for a specific host language, let's take a closer look at their structure and its constituent parts.
```